features/joinroom/api/src/main/kotlin/io/element/android/features/joinroom/api/JoinRoomEntryPoint.kt

```diff
@@ -22,13 +22,15 @@ import io.element.android.features.roomdirectory.api.RoomDescription
 import io.element.android.libraries.architecture.FeatureEntryPoint
 import io.element.android.libraries.architecture.NodeInputs
 import io.element.android.libraries.matrix.api.core.RoomId
+import io.element.android.libraries.matrix.api.core.RoomIdOrAlias
 import java.util.Optional
 
 interface JoinRoomEntryPoint : FeatureEntryPoint {
     fun createNode(parentNode: Node, buildContext: BuildContext, inputs: Inputs): Node
 
     data class Inputs(
         val roomId: RoomId,
+        val roomIdOrAlias: RoomIdOrAlias,
```

Comment:
I'd use an `Optional<RoomAlias>` instead

Reply:
That would be more correct, hence it brings some complexity.
At some point the room resolver should probably be moved in the `joinroom` module (I am more and more convinced about this approach), and so, the first param `roomId` should  just be removed.